test/integration/ssl_integration_test.cc

```diff
@@ -40,6 +44,7 @@ void SslIntegrationTest::TearDownTestCase() {
 }
 
 ServerContextPtr SslIntegrationTest::createUpstreamSslContext() {
+  static Stats::TestIsolatedStoreImpl upstream_stats_store;
```

Comment:
The problem is that if you share a stats store between the Envoy server instance and upstream instance, you get into a dependency cycle. Previously the call to `createUpstreamSslContext` happened after the `test_server_` was initialized. Now we need the test server to be created after the upstream, to allow the upstream to find its bound port and substitute the port into the JSON config prior to initializing the test server.

The test code doesn't check the stats for the upstream instances, only for the test server AFAICT.